engine/pkg/meta/internal/sqlkv/sql_impl.go

```diff
@@ -71,25 +75,31 @@ func NewSQLKVClientImpl(sqlDB *sql.DB, table string, jobID metaModel.JobID) (*sq
 
 	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
 	defer cancel()
-
 	impl := &sqlKVClientImpl{
 		db:           db,
 		jobID:        jobID,
 		tableScopeDB: tableScopeDB,
+		table:        table,
 	}
-	if err := impl.Initialize(ctx); err != nil {
+	if err := impl.initialize(ctx); err != nil {
 		return nil, err
 	}
 
 	return impl, nil
 }
 
-// Initialize initializes metakv table
-// NOTE: Make Sure to call InitializeEpochModel before Initialize any KVClient
-func (c *sqlKVClientImpl) Initialize(ctx context.Context) error {
-	if err := c.tableScopeDB.
-		WithContext(ctx).
-		AutoMigrate(&sqlkvModel.MetaKV{}); err != nil {
```

Comment:
An alternative way is to keep the `AutoMigrate` and ignore the `table already exists` error, which doesn't need to hard code create table DDL.

Reply:
It looks weird that `AutoMigrate` doesn't handle `table already exists` internally. 🤔

Reply:
> It looks weird that `AutoMigrate` doesn't handle `table already exists` internally. 🤔

It's the problem of implment. It first checks the existence of the table and then create the table. 
So it's designed to be irreentrance. May need some protection.

Reply:
I will check if `CREATE TABLE IF NOT EXISTS` is a common SQL.
If so, may be we can make a pr for the gorm.

Reply:
> 

Can we use a lock to protect the `AutoMigrate` function? It seems it is not thread-safe.

Reply:
The competition comes from other jobs, not the single job calls 'initialize' concurrently.

Reply:
The competition comes from other jobs, not the single job calls 'initialize' concurrently.

Reply:
If multiple jobs need the same table, is it better to create the table before creating `sqlKVClientImpl`? (Just for discussion)

Reply:
It's designed to create the meta kv table inside the 'creating client', because the table name is related to 'projectID' and I want to hide this implement internally.

Or just as you said, we can create the meta kv table in `JobMaster`.